devito/types/basic.py

```diff
@@ -1167,12 +1167,16 @@ def _evaluate(self, **kwargs):
         # Apply interpolation from inner most dim
         for d, i in self._grid_map.items():
             retval = retval.diff(d, deriv_order=0, fd_order=2, x0={d: i})
-        if self._avg_mode == 'harmonic':
-            retval = 1 / retval
 
         # Evaluate. Since we used `self.function` it will be on the grid when evaluate
         # is called again within FD
-        return retval.evaluate.expand()
+        if self._avg_mode == 'harmonic':
+            from devito.finite_differences.differentiable import SafeInv
+            retval = SafeInv(retval.evaluate, base=self.function)
+        else:
+            retval = retval.evaluate
```

Comment:
if there was an expand() before, I'd be surprised no tests failed now

Reply:
the `expand` was only there for the harmonic case to avoid nested `1/ (1/ (1/a + 1/ b)))` and such, now that it goes through it proper safe div it's not needed anymore